### PR TITLE
Upgrade node version

### PR DESCRIPTION
**Pull request by tuyuribr**
# PR Details
It's time to upgade the node version

## Description

The latest ubuntu version, installs node v18. So we can upgrade node and the other dependencies 

[https://packages.ubuntu.com/kinetic/nodejs]

## Warnings
Win7 don't support node 18 ... (only up to 13)

## Types of changes

- [x] Dependency change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## TO DO
- [ ] Quick steps for upgrade
- [ ] Other dependencies update
- [ ] User base test

## Tested
Windows. Node v18.16.0. WaWeb 2.2326.10. [2023-06-24]

### CODE REVIEW

**Comment by tuyuribr** [comment]
I think node v16 gives us a little more time to the next upgrade

**Comment by tuyuribr** [comment]
Which puppeteer version should we choose? What are the breaking changes that we should avoid?
I think we should upgrade to v18.2.1 
https://github.com/puppeteer/puppeteer/tree/v18.2.1

**Comment by tuyuribr** [comment]
I think we should upgrade node-fetch to v2.6.9 (nice). We are in 2.6.5 so probably won't break a thing.

https://github.com/node-fetch/node-fetch/releases/tag/v2.6.9

**Comment by amosayomide05** [comment]
I wonder why the library supports old version

**Comment by tuyuribr** [comment]
> I wonder why the library supports old version

Because the last version of ubuntu would install node v12 on apt install nodejs

**Comment by wictornogueira** [comment]
> I think node v16 gives us a little more time to the next upgrade

Sort of... v16 will be reaching EOL in 5 months

**Comment by wictornogueira** [comment]
Here 👇

### Windows

#### Manual
Just get the latest LTS from https://nodejs.org

#### Choco
```powershell
choco install nodejs-lts
```

#### Winget
```powershell
winget install OpenJS.NodeJS.LTS
```

### Ubuntu / Debian
```bash
curl -fsSL https://deb.nodesource.com/setup_lts.x | sudo -E bash - &&\
sudo apt-get install -y nodejs
```

File: README.md [hunk 1]
```diff
@@ -18,7 +18,9 @@ It uses Puppeteer to run a real instance of Whatsapp Web to avoid getting blocke
 
 The module is now available on npm! `npm i whatsapp-web.js`
 
-Please note that Node v12+ is required.
+Please note that Node v16+ is required.
+
+--QUICK STEPS TO INSTALL NEW VERSION
```
Comment (wictornogueira) [inline README.md:23]: Here 👇

### Windows

#### Manual
Just get the latest LTS from https://nodejs.org

#### Choco
```powershell
choco install nodejs-lts
```

#### Winget
```powershell
winget install OpenJS.NodeJS.LTS
```

### Ubuntu / Debian
```bash
curl -fsSL https://deb.nodesource.com/setup_lts.x | sudo -E bash - &&\
sudo apt-get install -y nodejs
```

**Review by wictornogueira — CHANGES_REQUESTED** [review]

**Comment by zvictor** [comment]
> > I wonder why the library supports old version
> 
> Because the last version of ubuntu would install node v12 on apt install nodejs

Thanks for sharing this info and for working on this PR! ❤️
I must say, it's quite surprising to learn that the decision to maintain an obsolete version was driven by Ubuntu's default Node.js installation.

Relying on Ubuntu as the primary deployment method seems arbitrary, particularly since this is just one of the many ways one could deploy WWeb applications. Moreover, it appears we haven't documented this decision or provided any deployment guidance, which means only those who happen to be already deploying on Ubuntu would benefit from it.

When I deployed my first app, I turned to Puppeteer's guides for recommended deployment steps. My deployment process still follows one of [their guides](https://pptr.dev/guides/docker).

I wholeheartedly agree with the changes proposed in this PR. As a Node project, rather than an Ubuntu one, it's important that we adhere to Node's release schedule. Sticking to whatever is the `Active LTS` version at the time is standard practice and will help prevent the accumulation of technical debt.

**Review by zvictor — APPROVED** [review]

**Comment by tuyuribr** [comment]
If you test this PR please include your node version ( `node --version` ), your system OS and the WaWeb version ( `await client.getWWebVersion()` )
Update your package.json to: git+https://github.com/pedroslopez/whatsapp-web.js#tuyuribr-patch-2
Don't forget to delete node_modules them npm i

**Comment by tofers** [comment]
13 Puppetmaster has stopped working properly.

Some files are not loading.

With Puppeteer 18.2.1 files are loaded successfully.  I will be switching

**Review by alechkos — APPROVED** [review]
Windows 10

Node v18.17.1
Puppeteer v18.2.1
WWeb v2.2333.11
WWebJS v1.22.1

Tested some library functions, version caching and my opened PRs, works perfect

**Comment by wictornogueira** [comment]
> Warnings: Win7 only supports node 13

Well, we could always add a note about that 🤷
I wouldn't be that concerned, though

**Comment by diegofemello** [comment]
up

**Review by zvictor — APPROVED** [review]
awesome

**Review by purpshell — APPROVED** [review]

**Comment by devagarwal007** [comment]
how to use this newly pushed code?

**Comment by washington65145as** [comment]
> how to use this newly pushed code?

do npm install github:pedroslopez/whatsapp-web.js#webpack-exodus in your path project

**Pull request merged**